File: packages/flutter_tools/test/src/base/logger_test.dart
// Copyright 2017 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter_tools/src/base/logger.dart';
import 'package:flutter_tools/src/base/os.dart';
import 'package:flutter_tools/src/base/version.dart';
import 'package:test/test.dart';

void main() {
  group('AnsiLogger', () {
    test('should support color on Windows version >= 10.0.10586', () {
      final String name = new OperatingSystemUtils().name;
      final String versionString = new RegExp(r'\d+\.\d+\.\d+').firstMatch(name).group(0);
      final Version version = new Version.parse(versionString);
      final bool shouldSupportColor = version >= new Version.parse('10.0.10586');

      expect(new AnsiTerminal().supportsColor, shouldSupportColor);
    }, testOn: 'windows');
  });
}

File: engine/src/flutter/shell/platform/darwin/common/framework/Source/LoggerTest.swift
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import Foundation
import Testing

import InternalFlutterSwiftCommon
import test_utils_swift

@Suite struct LoggerTest {

  @Test func testInitialization() {
    let writer = StringOutputWriter()
    let logger = Logger(outputWriter: writer, logLevel: .info)
    #expect(logger.logLevel == .info)
  }

  @Test func testDoesNotLogMessageBelowLogLevel() {
    let writer = StringOutputWriter()
    let logger = Logger(outputWriter: writer, logLevel: .warning)

    logger.log(level: .info, "Hello world")
    #expect(!writer.didLog)
  }

  @Test func testLogsMessageAtLogLevel() {
    let writer = StringOutputWriter()
    let logger = Logger(outputWriter: writer, logLevel: .info)

    logger.log(level: .info, "Hello world")
    #expect(writer.didLog)
    #expect(writer.lastLevel == .info)
    #expect(writer.lastLine == "Hello world")
  }

  @Test func testLogsMessageAboveLevel() {
    let writer = StringOutputWriter()
    let logger = Logger(outputWriter: writer, logLevel: .info)

    logger.log(level: .warning, "Hello world")
    #expect(writer.didLog)
    #expect(writer.lastLevel == .warning)
    #expect(writer.lastLine == "Hello world")
  }

  // Verify that important messages are logged, even if log level is set to error.
  @Test func testLogsImportant() {
    let writer = StringOutputWriter()
    let logger = Logger(outputWriter: writer, logLevel: .error)

    logger.log(level: .important, "Hello world")
    #expect(writer.didLog)
    #expect(writer.lastLevel == .important)
    #expect(writer.lastLine == "Hello world")
  }

}
